pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -130,7 +127,20 @@ func readInstanceID(cfg *VSphereConfig) (string, error) {
 
 	s := object.NewSearchIndex(c.Client)
 
-	svm, err := s.FindByUuid(ctx, dc, strings.ToLower(strings.TrimSpace(out.String())), true, nil)
+	var svm object.Reference
+	for _, v := range addrs {
+		// Strip the cidr
+		ip := strings.Split(v.String(), "/")[0]
```

Comment:
it feels wrong to manually parse the cidr with strings. Why not use net.ParseCIDR()?